### PR TITLE
bump ginkgo to v2

### PR DESCRIPTION
**Pull request by lorrielau**
#### What this PR does / why we need it:
Bump ginkgo to v2.

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:

@Congrool 
#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: pkg/util/kubernetes/kubeadm/app/util/initsystem/initsystem_unix.go [hunk 1]
```diff
@@ -1,3 +1,4 @@
+//go:build !windows
```
Comment (Congrool) [inline pkg/util/kubernetes/kubeadm/app/util/initsystem/initsystem_unix.go:1]: this tag is used in go1.17 or above. We're in 1.16 currently, so do not update it.

File: pkg/util/kubernetes/kubeadm/app/util/runtime/runtime_unix.go [hunk 2]
```diff
@@ -1,3 +1,4 @@
+//go:build !windows
```
Comment (Congrool) [inline pkg/util/kubernetes/kubeadm/app/util/runtime/runtime_unix.go:1]: ditto

File: pkg/util/kubernetes/kubeadm/app/constants/constants_unix.go [hunk 3]
```diff
@@ -1,3 +1,4 @@
+//go:build !windows
```
Comment (Congrool) [inline pkg/util/kubernetes/kubeadm/app/constants/constants_unix.go:1]: Please remove this tag.

Comment (lorrielau) [reply pkg/util/kubernetes/kubeadm/app/constants/constants_unix.go]: Will do. :>

File: test/e2e/yurt/yurt.go [hunk 4]
```diff
@@ -54,11 +54,13 @@ func Register() {
 
 		err = ns.DeleteNameSpace(c, YurtE2ENamespaceName)
 		util.ExpectNoError(err)
-
-		ginkgo.By(YurtE2ETestDesc + "yurt_test_create namespace")
-		_, err = ns.CreateNameSpace(c, YurtE2ENamespaceName)
-		gomega.Expect(err).NotTo(gomega.HaveOccurred(), "fail to create namespace")
-
+		util.YurtDescribe(YurtE2ETestDesc+"yurt_test_create namespace", func() {
```
Comment (Congrool) [inline test/e2e/yurt/yurt.go:57]: Should it be BeforeEach rather than Describe?
We need to create pod in this namespace later in test `pod_operate`.

Comment (Congrool) [reply test/e2e/yurt/yurt.go:57]: In this manner, we depend on that `yurt_test_create namespace` should run before the `pod_operate`. It is not always the case.

**Comment by lorrielau** [comment]
@Congrool  Please take a look. Sorry for the late feedback. I've deleted the ginkgo.by part in e2e-yurt-create-namespace.

File: test/e2e/e2e.go [hunk 5]
```diff
@@ -37,14 +37,19 @@ import (
...
+			reporters.GenerateJUnitReport(r, path.Join(yurtconfig.YurtE2eCfg.ReportDir, fmt.Sprintf("yurt-e2e-test-report_%02d.xml", types.NewDefaultSuiteConfig().ParallelProcess)))
+			//r = append(r, reporters.NewJUnitReporter(path.Join(yurtconfig.YurtE2eCfg.ReportDir, fmt.Sprintf("yurt-e2e-test-report_%02d.xml", config.GinkgoConfig.ParallelNode))))
 		}
 	}
-	klog.Infof("Starting e2e run %q on Ginkgo node %d", uuid.NewUUID(), config.GinkgoConfig.ParallelNode)
-	ginkgo.RunSpecsWithDefaultAndCustomReporters(t, "openyurt e2e suite", r)
+	klog.Infof("Starting e2e run %q on Ginkgo node %d", uuid.NewUUID(), types.NewDefaultSuiteConfig().ParallelProcess)
+	//klog.Infof("Starting e2e run %q on Ginkgo node %d", uuid.NewUUID(), config.GinkgoConfig.ParallelNode)
+	ginkgo.RunSpecs(t, "openyurt e2e suite")
+	//ginkgo.RunSpecsWithDefaultAndCustomReporters(t, "openyurt e2e suite", r)
```
Comment (Congrool) [inline test/e2e/e2e.go:54]: Please remove these comments

File: test/e2e/e2e.go [hunk 6]
```diff
@@ -37,14 +37,19 @@ import (
...
-	var r []ginkgo.Reporter
+	//var r []ginkgo.Reporter
+	var r types.Report
+	//var rc types.ReporterConfig
 	if yurtconfig.YurtE2eCfg.ReportDir != "" {
 		if err := os.MkdirAll(yurtconfig.YurtE2eCfg.ReportDir, 0755); err != nil {
 			klog.Errorf("Failed creating report directory: %v", err)
 		} else {
-			r = append(r, reporters.NewJUnitReporter(path.Join(yurtconfig.YurtE2eCfg.ReportDir, fmt.Sprintf("yurt-e2e-test-report_%02d.xml", config.GinkgoConfig.ParallelNode))))
+			reporters.GenerateJUnitReport(r, path.Join(yurtconfig.YurtE2eCfg.ReportDir, fmt.Sprintf("yurt-e2e-test-report_%02d.xml", types.NewDefaultSuiteConfig().ParallelProcess)))
```
Comment (Congrool) [inline test/e2e/e2e.go:47]: Check the err returned by `reporters.GenerateJUnitReport`.

**Comment by Congrool** [comment]
Please rebase and squash your commits.

**Review by rambohe-ch — APPROVED** [review]

**Pull request merged**